astropy/logger.py

```diff
@@ -1,5 +1,12 @@
 # Licensed under a 3-clause BSD style license - see LICENSE.rst
-"""This module defines a logging class based on the built-in logging module"""
+"""This module defines a logging class based on the built-in logging module.
+
+.. note::
+
+    This module is meant for ``astropy`` usage.
+    If you wish to use it for your own packages, your mileage might vary.
```

Comment:
Maybe we want to be more explicit? For example,
```suggestion
    This module is meant for internal ``astropy`` usage.
    For use in other packages, we recommend implementing your own logger class
    and possibly using this implementation as a template.
```

Reply:
I would not people make a class, because then you need to set the logger class before instantiating a logger (https://github.com/astropy/astropy/blob/838d92e4bc03d9ca992ed327b986d87890106413/astropy/logger.py#L97-L103), which then leads to instantiating a global logger because it's pain. logging is not meant to be used like this.

Reply:
Is the suggested text okay if I remove "class" from the words?

Reply:
Yeah, I would also remove the end of the sentence ;)

Reply:
Okay, I updated the text. What about now?

Reply:
Yes, works for me !